server/controllers/thoughtController.js

```diff
@@ -0,0 +1,127 @@
+const User = require("../model/User");
+const Thought = require("../model/Thought");
+
+const addThought = async (req, res) => {
+    try {
+        const thought = await Thought.create(req.body);
+        const user = await User.findOneAndUpdate(
+            { username: req.body.username },
+            { $addToSet: { thoughts: thought._id } },
+            { new: true }
+        )
+        res.status(200).json({ user, thought });
+    } catch (err) {
+        console.log(err);
+        res.status(500).json(err);
+    }
+}
+
+const getAllThoughts = async (req, res) => {
+    try {
+        const thoughts = await Thought.find();
+        res.status(200).json(thoughts);
+    } catch (err) {
+        console.log(err);
+        res.status(500).json(err);
+    }
+}
+
+const getThoughtById = async (req, res) => {
+    const _id = req.params.id;
+    try {
+        const thought = await Thought.findOne({ _id })
+        if (!thought) {
+            return res.status(404).json({ message: 'No thought with that ID' })
```

Comment:
This handling of edge case scenarios is super important in creaitng robust apps, consider taking it forward in future work :-)